iroh-api/src/store.rs

```diff
@@ -19,20 +20,43 @@ pub trait Store: 'static + Send + Sync + Clone {
     async fn put_many(&self, blocks: Vec<Block>) -> Result<()>;
 }
 
+#[derive(Debug, Clone)]
+pub struct StoreApi {
+    client: StoreClient,
+}
+
+impl StoreApi {
+    pub fn new(client: StoreClient) -> Self {
+        Self { client }
+    }
+
+    pub async fn has(&self, cid: Cid) -> Result<bool> {
+        self.client.has(cid).await
+    }
+
+    pub async fn put(&self, cid: Cid, blob: Bytes, links: Vec<Cid>) -> Result<()> {
+        self.client.put(cid, blob, links).await
+    }
+
+    pub async fn put_many(&self, blocks: Vec<Block>) -> Result<()> {
+        self.client
+            .put_many(blocks.into_iter().map(|x| x.into_parts()).collect())
+            .await
+    }
+}
```

Comment:
I'm a bit confused why we have both pub methods here as well as the trait?  I guess I'd be happy removing the trait?  I'm not sure what it gives us?